tests/test_mock_trie.c

```diff
@@ -0,0 +1,97 @@
+#include <criterion/criterion.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include "mock_trie.h"
+#include <string.h>
+/* Checks whether a new trie is properly created */
+Test(trie, new)
+{
+    trie_t *new_trie;
+    char** word = malloc(sizeof(char *) * 70);
+
+    new_trie = trie_new(word);
+
+    cr_assert_not_null(new_trie, "trie_new() failed");
+}
+
+/* Checks whether a trie is initiated properly */
+Test(trie, init)
+{
+    int rc;
+    trie_t new_trie;
+  //  char** words = malloc(sizeof(char *) * 70);
+
+    rc = trie_init(&new_trie);
+
+    cr_assert_eq(rc, 1, "trie_init() failed");
+
+}
+
+/* Checks whether the trie is correctly freed */
+Test(trie, free)
+{
+    trie_t *new_trie;
+    char** wordss = malloc(sizeof(char *) * 70);
+    int rc;
+
+    new_trie = trie_new(wordss);
+
+    cr_assert_not_null(new_trie, "trie_new() failed");
+
+    rc = trie_free(new_trie);
+
+    cr_assert_eq(rc, 1, "trie_free() failed");
+}
+
+/* Checks the case in which a new word is being added to the trie*/
+Test(trie, added_to_trie){
+  trie_t *trie1 = trie_new();
+  char *s1 = (char *)malloc(sizeof(char) * 3);
+  s1 = "hi";
+  int added_to_trie = add_to_trie(s1, trie1);
+  cr_assert_eq(added_to_trie, 1, "add_to_trie failed");
+}
+
+/* Checks the case in which the same word is being tried to add to the trie */
+Test(trie, same_variable_add_to_trie){
+  trie_t *trie2 = trie_new();
+  char *ss1 = (char *)malloc(sizeof(char) * 3);
+  ss1 = "hi";
+  add_to_trie(ss1, trie2);
+  int same_word_add_to_trie2 = add_to_trie(ss1, trie2);
+  cr_assert_eq(same_word_add_to_trie2, 0, "add_to_trie failed");
+}
+
+/* Checks the case in which the word cannot be added to the trie */
+Test(trie, cannot_add_to_trie){
+  trie_t *trie3 = trie_new();
+  char *sss1 = (char *)malloc(sizeof(char) * 53);
+  sss1 = "nojfkdsjfksdjfdsjfkdsjfjfkdsjfkdsjfkdsjkfiejkdjskfsd";
+  int cant_add_to_trie = add_to_trie(sss1, trie3);
+  cr_assert_eq(cant_add_to_trie, 1, "add_to_trie failed");
+}
+
+/* Checks the case in which the word is in the trie */
+Test(Trie, in_trie) {
```

Comment:
Most of your trie tests are in the `trie` suite, but this and the one below are in the `Trie` suite. This results in the following:
```
$ ./tests/test-spellcheck -l
Trie: 2 tests
├── in_trie
└── not_in_trie
dictionary: 21 tests
├── add_to_dict_empty
├── add_to_dict_f0
├── add_to_dict_f1
├── add_to_dict_null0
├── add_to_dict_null1
├── add_to_dict_s0
├── add_to_dict_s1
├── free
├── in_dict_empty
├── in_dict_f0
├── in_dict_f1
├── in_dict_null0
├── in_dict_null1
├── in_dict_s0
├── in_dict_s1
├── init
├── new
├── read_to_dict0
├── read_to_dict_noname
├── read_to_dict_null0
└── read_to_dict_null1
trie: 6 tests
├── added_to_trie
├── cannot_add_to_trie
├── free
├── init
├── new
└── same_variable_add_to_trie
```
This is no good; the trie tests should all reside in the same suite.

Reply:
Also, it appears that the `dictionary` suite runs before the `trie` suite, which is no good since you've built the dictionary on top of the trie. The `trie` suite needs to run first; enforce that.

Reply:
And while we're talking about tests, when I run them (on `linux.cs`), I get the following:
```
$ tests/test-spellcheck 
src/dictionary.c::add_to_dict::74 Invalid arguments to add_to_dict
src/dictionary.c::add_to_dict::74 Invalid arguments to add_to_dict
src/dictionary.c::in_dict::64 Invalid input to in_dict
src/dictionary.c::in_dict::64 Invalid input to in_dict
src/dictionary.c::in_dict::64 Invalid input to in_dict
src/dictionary.c::add_to_dict::74 Invalid arguments to add_to_dict
src/dictionary.c::add_to_dict::74 Invalid arguments to add_to_dict
[====] Synthesis: Tested: 29 | Passing: 29 | Failing: 0 | Crashing: 0
```
Successful tests should result in clean output (e.g. only the last line). Either the tests aren't correct (and they're passing when they shouldn't be), or some error detection is incorrect (and is printing errors when it shouldn't be).